Make api gateway optional and adjust api_base_auth var naming

Erin-Boehmer opened the pull request:
Addresses issue https://github.com/outerbounds/terraform-aws-metaflow/issues/22

In production environments, some teams will not want to use public-facing API Gateway resources. This PR makes API Gateway optional by adding a var to the metadata service `enable_api_gateway`. I also adjusted the `api_basic_auth` var to use the `enable_` prefix, in keeping with the bool var naming in this repo.

oavdeev reviewed (approved):
This change would technically require the module user to change the param name, right? Just making note of this so i'll make it a minor release then, not a patch release

Erin-Boehmer commented:
> This change would technically require the module user to change the param name, right? Just making note of this so i'll make it a minor release then, not a patch release

Hey @oavdeev - yes, this would be a minor release. I refactored the `api_basic_auth` to `enable_api_basic_auth` (within the metadata service module) and `metadata_service_enable_api_basic_auth` (at the metaflow module level). Also rebased against latest commits if you're ready to merge.
